### PR TITLE
Block list

### PR DESCRIPTION
**Pull request by takafumi-hino**
public dir is not changed.
src dir has some changed.
 |
 L___detail.js :  is not changed. Since this program may fall into an infinite loop, it will be pushed after confirmation and correction.
 L___index.css : is not changed. Used to list blocks.
 L___index.js : Displays a list of blocks. Will be renamed to list.js in the future.
 L___server.js  : is not changed. Separated from server2.js for future routing implementation. 
 L___server2.js : Pass block information together. it is separeted from server.js

### CODE REVIEW

File: src/server2.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  let sha256 = new jsSHA('SHA-256', 'HEX');
+  sha256.update(text);
+  return sha256.getHash('HEX');
+}
+
+app.get('/list/:linesPerPage', (req, res) => {
+
+  const regex = new RegExp(/^[0-9]{3}$/);
+ 
+  let linesPerPage = +req.params.linesPerPage;
```
Comment (rantan) [inline src/server2.js]: `+` is for what?

File: src/server2.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  let promiseArray = [];
+
+  getBlockchainInfo((blockchainInfo) => {
+
+    let bestBlockHash = blockchainInfo.bestblockhash;
+    let bestBlockHeight = blockchainInfo.headers;
+    
+    elect.request('blockchain.block.headers',[bestBlockHeight-linesPerPage+1,linesPerPage,0],async function(err,rep){
```
Comment (rantan) [inline src/server2.js]: Can you add `page` parameter for to specify page number of response.

File: src/server2.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      
+      if(err) throw err
+
+      let headersHex = rep.result.hex;
+      let headerHex = headersHex.match(/.{160}/g);
+      let header = [];
+            
+      for(let i=0; i<linesPerPage; i++){
+        let calcHash = sha256(sha256(headerHex[i]));
+        let byteOrder = calcHash.match(/.{2}/g);
```
Comment (rantan) [inline src/server2.js]: Can you extract this codes which is for changing byte order as a function.

File: src/server2.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+app.get('/list/:linesPerPage', (req, res) => {
+
+  const regex = new RegExp(/^[0-9]{3}$/);
+ 
+  let linesPerPage = +req.params.linesPerPage;
+  let promiseArray = [];
+
+  getBlockchainInfo((blockchainInfo) => {
```
Comment (rantan) [inline src/server2.js]: Can you use `async/await` syntax in where it can be used like here?

File: src/server2.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    
+    elect.request('blockchain.block.headers',[bestBlockHeight-linesPerPage+1,linesPerPage,0],async function(err,rep){
+      
+      if(err) throw err
+
+      let headersHex = rep.result.hex;
+      let headerHex = headersHex.match(/.{160}/g);
+      let header = [];
+            
+      for(let i=0; i<linesPerPage; i++){
```
Comment (rantan) [inline src/server2.js]: You can use `forEach` or `map` to simplify loop.

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+class App extends React.Component{
+  constructor(props,context){
+    super(props,context)
+    this.state = {
+      data :  []
+    }
+  }
+
+  getBlockInfo = () => {
+    axios.get(`${"http://localhost:3001/list"}/${linesPerPage}`).then(res=>{
```
Comment (rantan) [inline src/index.js]: I think you can use `async/await` syntax here. Can you try?

File: src/index.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+      for(let i=0; i<linesPerPage; i++){
```
Comment (rantan) [inline src/index.js]: This loop can be eliminate.

File: src/index.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+class App extends React.Component{
+  constructor(props,context){
+    super(props,context)
+    this.state = {
+      data :  []
+    }
+  }
+
+  getBlockInfo = () => {
```
Comment (rantan) [inline src/index.js]:
```suggestion
  getBlockInfo() {
```

**Comment by rantan** [comment]
Can you add config for github actions to check eslint when PRs created?

File: src/server2.js (hunk 1 above)
Comment (takafumi-hino) [reply src/server2.js]: Created with the intention of obtaining the number of items to be displayed in the list from the URL. But I don't need it now so delete it.

File: src/server2.js (hunk 3 above)
Comment (takafumi-hino) [reply src/server2.js]: resolved.

File: src/index.js (hunk 8 above)
Comment (takafumi-hino) [reply src/index.js]: resolved

File: src/index.js (hunk 7 above)
Comment (takafumi-hino) [reply src/index.js]: resolved.

File: src/index.js (hunk 6 above)
Comment (takafumi-hino) [reply src/index.js]: resolved

**Pull request closed without merging**